File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.cs
public class Solution {
    public int LongestOnes(int[] nums, int k) {
        int maxLength = 0, left = 0, zeroCnt = 0;
        for(int i = 0; i < nums.Length; i++){
            if(nums[i] == 0) zeroCnt++;
            
            while(zeroCnt > k){
                if(nums[left] == 0) zeroCnt--;
                left++;                
            }
            maxLength = Math.Max(maxLength, i-left + 1);
        }        
        return maxLength;
        
    }
}

File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.ts
function longestOnes(nums: number[], k: number): number {
    let l: number = 0;
    let r: number = 0;
    let ret: number = 0;
    let zeroCount = 0;
    
    while(r < nums.length){
        if(nums[r] === 0){
            zeroCount++;
            while(zeroCount > k){
                if(nums[l] === 0)
                    zeroCount--;                
                l++;
            }            
        }
        r++;
        ret = Math.max(r-l, ret);
    }
    
    return ret;
};